src/plugins/data/public/ui/query_string_input/query_bar_top_row.tsx

```diff
@@ -65,11 +73,47 @@ interface Props {
   timeHistory?: TimeHistoryContract;
 }
 
+function showNotification(
+  notifications?: CoreStart['notifications'],
+  data?: DataPublicPluginStart
+) {
+  return notifications!.toasts.addInfo(
+    {
+      title: i18n.translate('data.query.queryBar.longQuery', {
+        defaultMessage: 'Your query is taking a while',
+      }),
+      text: toMountPoint(<LongQueryNotification data={data} />),
+    },
+    { toastLifeTimeMs: 100000 }
+  );
+}
+
 export function QueryBarTopRow(props: Props) {
   const [isDateRangeInvalid, setIsDateRangeInvalid] = useState(false);
 
   const kibana = useKibana<IDataPluginServices>();
-  const { uiSettings, notifications, storage, appName, docLinks } = kibana.services;
+  const { uiSettings, notifications, storage, appName, docLinks, data } = kibana.services;
+  useEffect(() => {
+    let timer: any;
+    let toast: Toast;
+    const searchObservable = data.search.getPendingCount$();
+    const subscription = searchObservable?.subscribe((count: number) => {
+      if (count > 0) {
+        if (timer) clearTimeout(timer);
```

Comment:
I don't think this is going to work. `getPendingCount$` emits even when the number of pending requests goes down, so we're really going to be clearing the timeout & rescheduling even when requests come back. Also, even if this weren't the case, we don't want to unschedule the notification when new requests happen because we really want any request that takes longer than 10s (or whatever we determine) to be able to bring up the popup.

Not sure if that makes sense, but here's an example:

1. Search request #1 goes out (notification is scheduled)
1. After 5 seconds, search request #2 goes out (notification is rescheduled)
1. After 5 more seconds, search request #1 has been running for 10s but the notification doesn't pop-up

This might not seem like a problem but when you've got a bunch of requests happening serially then you can eventually get up past the 30s timeout before you even show the notification.